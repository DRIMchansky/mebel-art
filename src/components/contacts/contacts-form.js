// packages
import React, { useState } from 'react'
import styled from 'styled-components'
import cogoToast from 'cogo-toast'
import axios from 'axios'
// compoents
import { Grid, TextField } from '@material-ui/core'
import CustomButton from '../custom-button'
import Title from '../title'
import SocialHorizontal from '../social-horizontal'
// icons
import SendIcon from '@material-ui/icons/Send'
import HourglassEmptyIcon from '@material-ui/icons/HourglassEmpty'
// utils
import { colors, shadows, breakpoints } from '../../util/css-config'

const ContactsForm = () => {
  const [formName, setFormName] = useState('')
  const [formPhone, setFormPhone] = useState('')
  const [formMessage, setFormMessage] = useState('')

  const [loading, setLoading] = useState(false)

  const handleSubmit = (e) => {
    e.preventDefault()
    const data = new FormData(e.target)
    setLoading(true)

    axios({
      method: 'POST',
      url: 'https://formsubmit.co/ajax/deko32@mail.ru',
      // url: 'https://formsubmit.co/ajax/drimchansky@gmail.com',
      data: data,
    })
      .then(function (response) {
        cogoToast.success('Ваше сообщение отправлено!')
        setFormName('')
        setFormPhone('')
        setFormMessage('')
        setLoading(false)
      })
      .catch(function (response) {
        cogoToast.error(
          'Что-то пошло не так, вы можете связаться с нами через социальные сети или по телефону',
          {
            hideAfter: 10,
          }
        )
      })
  }

  return (
    <FormStyled onSubmit={handleSubmit} id="contact-form">
      <Title alignment="left">
        <h2>Заинтересованы в покупке кухни мечты?</h2>
        <span>Вы можете связаться с нами в удобной для вас социальной сети или мессенджере:</span>
      </Title>
      <SocialHorizontal />
      <Title alignment="left">
        <span>Или оставить заявку на звонок:</span>
      </Title>
      <Grid container spacing={2} justify="center">
        <Grid item container xs={12} sm={8} md={5} justify="center">
          <TextFieldStyled
            id="name"
            label="Имя"
            name="Имя"
            onChange={(e) => setFormName(e.target.value)}
            value={formName}
            color="primary"
            variant="outlined"
            required
            fullWidth
            style={{
              marginRight: '3px',
            }}
          />
          <TextFieldStyled
            id="tel"
            type="tel"
            label="Телефон"
            name="Телефон"
            onChange={(e) => setFormPhone(e.target.value)}
            value={formPhone}
            variant="outlined"
            color="primary"
            margin="normal"
            required
            fullWidth
            style={{
              marginRight: '3px',
            }}
          />
        </Grid>
        <Grid item container xs={12} sm={8} md={7}>
          <TextFieldStyled
            id="textarea"
            multiline
            label="Cообщение"
            name="Сообщение"
            onChange={(e) => setFormMessage(e.target.value)}
            value={formMessage}
            variant="outlined"
            color="primary"
            required
            fullWidth
            style={{
              marginRight: '3px',
            }}
          />
        </Grid>
        <Grid item container xs={12} justify="center">
          <CustomButton
            bgcolor={colors.dark}
            textcolor={colors.white}
            size="large"
            type="submit"
            variant="contained"
            startIcon={loading ? <HourglassEmptyIcon /> : <SendIcon />}
            margintop="1.5rem"
            disabled={loading}>
            Отправить
          </CustomButton>
        </Grid>
      </Grid>
    </FormStyled>
  )
}

const FormStyled = styled.form`
  margin: 0 auto;
  padding: 1rem 0.5rem 2rem;

  @media (min-width: ${breakpoints.medium}) {
    position: absolute;
    bottom: -2rem;
    left: 50%;
    transform: translate(-50%, 3%);
    padding: 3rem 3.5rem 4rem 3.5rem;
    width: 900px;
    background: ${colors.white};
    box-shadow: ${shadows.five};
    border-radius: 7px;
  }

  @media (min-width: ${breakpoints.large}) {
    transform: translate(-50%, 33%);
  }
`
//
//  Overwriting MaterialUI styles. Using IMPORTANT is justified
//

const TextFieldStyled = styled(TextField)`
  flex-direction: row !important;

  @media (min-width: ${breakpoints.medium}) {
    margin-bottom: 0 !important;
  }

  & fieldset {
    border-color: ${colors.black} !important;
  }

  & label {
    color: ${colors.black} !important;
    @media (min-width: 960px) {
      color: ${colors.black} !important;
    }
  }

  & textarea,
  & input {
    color: ${colors.black};
    height: 100% !important;

    @media (min-width: 960px) {
      color: ${colors.black};
      margin-bottom: none !important;
    }
  }
`

export default ContactsForm
