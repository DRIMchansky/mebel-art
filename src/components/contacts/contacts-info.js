// packages
import React from 'react'
import styled from 'styled-components'
import { graphql, useStaticQuery } from 'gatsby'
// icons
import PlaceIcon from '@material-ui/icons/Place'
import MailOutlineIcon from '@material-ui/icons/MailOutline'
import PhoneIphoneIcon from '@material-ui/icons/PhoneIphone'
// utils
import { colors } from '../../util/css-config'
import generateProtocolLink from '../../util/generate-protocol-link'

const ContactsInfo = () => {
  const data = useStaticQuery(graphql`
    query {
      allContentfulGeneralInfo {
        edges {
          node {
            eMail
            phone
          }
        }
      }
    }
  `).allContentfulGeneralInfo.edges[0].node

  return (
    <InfoStyled>
      <ul>
        <li>
          <PlaceIcon />
          <span>Где мы работаем</span>
          <small>Ярославль и Ярославская область</small>
        </li>
        <li>
          <MailOutlineIcon />
          <span>
            <a href={generateProtocolLink('email', data.phone)}>{data.eMail}</a>
          </span>
        </li>
        <li>
          <PhoneIphoneIcon />
          <span>
            <a href={generateProtocolLink('phone', data.phone)}>{data.phone}</a>
          </span>
        </li>
      </ul>
    </InfoStyled>
  )
}

const InfoStyled = styled.div`
  color: ${colors.dark};
  padding: 2rem;
  padding-left: 3rem;

  @media (min-width: 960px) {
    padding: 0;
    height: 100%;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: flex-start;
  }

  & ul {
    list-style-type: none;
    padding: 0;

    @media (min-width: 960px) {
      margin-top: 5rem;
    }
  }

  & li {
    position: relative;
    margin-top: 1rem;

    @media (min-width: 960px) {
      margin-top: 2rem;
    }
  }

  & span {
    font-weight: 400;
    display: block;
    font-size: 1.1rem;
    margin-bottom: 0.3rem;
    color: ${colors.black};

    & a {
      color: ${colors.black};
    }

    @media (min-width: 960px) {
      margin-bottom: 1rem;
    }
  }

  & small {
    font-weight: 300;
    font-size: 1rem;
    opacity: 0.8;
  }

  & svg {
    position: absolute;
    left: -2rem;
    top: -0.2rem;
    fill: ${colors.dark};
  }
`

export default ContactsInfo
