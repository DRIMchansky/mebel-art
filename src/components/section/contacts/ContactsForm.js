// packages
import React, { useState } from 'react'
import styled from 'styled-components'
import cogoToast from 'cogo-toast'
import axios from 'axios'
// compoents
import { Grid } from '@material-ui/core'
import TextField from '@material-ui/core/TextField'
import CustomButton from '../../ui/CustomButton'
// icons
import SendIcon from '@material-ui/icons/Send'
// utils
import { colors, shadows } from '../../../util/cssConfig'

const ContactsForm = () => {
  const [formName, setFormName] = useState('')
  const [formPhone, setFormPhone] = useState('')
  const [formMessage, setFormMessage] = useState('')

  const handleSubmit = (e) => {
    e.preventDefault()
    const data = new FormData(e.target)

    axios({
      method: 'POST',
      url: 'https://formsubmit.co/ajax/drimchansky@mail.com',
      data: data,
    })
      .then(function (response) {
        cogoToast.success('Ваше сообщение отправлено!')
        setFormName('')
        setFormPhone('')
        setFormMessage('')
      })
      .catch(function (response) {
        cogoToast.error('Что-то пошло не так')
      })
  }

  return (
    <FormStyled onSubmit={handleSubmit}>
      <TitleStyled>Заинтересованы в покупке кухни мечты?</TitleStyled>
      <Grid container spacing={2} justify="center">
        <Grid item container xs={12} sm={8} md={5} justify="center">
          <TextFieldStyled
            id="name"
            label="Имя"
            name="Name"
            onChange={(e) => setFormName(e.target.value)}
            value={formName}
            color="primary"
            variant="outlined"
            required
            fullWidth
            style={{
              marginRight: '3px',
            }}
          />
          <TextFieldStyled
            id="tel"
            type="tel"
            label="Телефон"
            name="phone"
            onChange={(e) => setFormPhone(e.target.value)}
            value={formPhone}
            variant="outlined"
            color="primary"
            margin="normal"
            required
            fullWidth
            style={{
              marginRight: '3px',
            }}
          />
        </Grid>
        <Grid item container xs={12} sm={8} md={7}>
          <TextFieldStyled
            id="textarea"
            multiline
            label="Cообщение"
            name="message"
            onChange={(e) => setFormMessage(e.target.value)}
            value={formMessage}
            variant="outlined"
            color="primary"
            required
            fullWidth
            style={{
              marginRight: '3px',
            }}
          />
        </Grid>
        <Grid item container xs={12} justify="center">
          <CustomButton
            bgcolor={colors.accent}
            textcolor={colors.white}
            size="large"
            type="submit"
            variant="contained"
            startIcon={<SendIcon />}
            margintop="1.5rem">
            Отправить
          </CustomButton>
        </Grid>
      </Grid>
    </FormStyled>
  )
}

const TitleStyled = styled.span`
  display: block;
  padding: 0.5rem 0 2rem 0;
  font-size: 1.2rem;
  color: ${colors.black};

  @media (min-width: 960px) {
    font-size: 1.4rem;
    margin-bottom: 1.5rem;
    color: ${colors.black};
  }
`

const FormStyled = styled.form`
  margin: 0 auto;
  padding: 1rem 0.5rem 2rem;

  @media (min-width: 960px) {
    position: absolute;
    bottom: -2rem;
    left: 50%;
    transform: translateX(-50%);
    padding: 3rem 3.5rem 4rem 3.5rem;
    width: 900px;
    background: ${colors.white};
    /* border: 1px solid black; */
    box-shadow: ${shadows.five};
    border-radius: 7px;
  }
`
//
//  Overwriting MaterialUI styles. Using IMPORTANT is justified
//

const TextFieldStyled = styled(TextField)`
  flex-direction: row !important;

  @media (min-width: 960px) {
    margin-bottom: 0 !important;
  }

  & fieldset {
    border-color: ${colors.black} !important;
  }

  & label {
    color: ${colors.black} !important;
    @media (min-width: 960px) {
      color: ${colors.black} !important;
    }
  }

  & textarea,
  & input {
    color: ${colors.black};
    height: 100% !important;

    @media (min-width: 960px) {
      color: ${colors.black};
      margin-bottom: none !important;
    }
  }
`

export default ContactsForm
