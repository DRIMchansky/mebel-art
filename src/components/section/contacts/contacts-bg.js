// packages
import React from 'react'
import styled from 'styled-components'
import { graphql, useStaticQuery } from 'gatsby'
// components
import BackgroundImage from 'gatsby-background-image'
// utils
import { colors } from '../../../util/css-config'

const ContactsBg = () => {
  const data = useStaticQuery(graphql`
    query {
      file(sourceInstanceName: { eq: "images" }, name: { eq: "kitchen-vertical" }) {
        childImageSharp {
          fluid {
            ...GatsbyImageSharpFluid_withWebp
          }
        }
      }
    }
  `)
  const imageData = data.file.childImageSharp.fluid
  return (
    <BG Tag="section" fluid={imageData} backgroundColor={`${colors.dark}`}>
      <Overflow />
    </BG>
  )
}

const BG = styled(BackgroundImage)`
  display: none;

  @media (min-width: 960px) {
    display: block;
    background: #000;
    height: 100%;
    height: 1000px;
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center center;
  }
`

const Overflow = styled.div`
  mix-blend-mode: multiply;
  height: 100%;
  width: 100%;
  background: ${colors.black};
  opacity: 0.5;
`

export default ContactsBg
