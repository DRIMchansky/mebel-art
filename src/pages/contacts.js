// packages
import React from 'react'
// components
import Layout from '../components/layout/layout'
import SEO from '../components/SEO'
import Contacts from '../components/section/contacts/contacts'

export default function contacts() {
  return (
    <>
      <SEO title="Наши контакты" description="Контакты т форма обратной связи" />
      <Layout>
        <Contacts />
      </Layout>
    </>
  )
}
