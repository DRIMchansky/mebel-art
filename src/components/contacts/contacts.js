// libraries
import React from 'react'
import styled from 'styled-components'
import { Grid, Container } from '@material-ui/core'
// components
import ContactsForm from './contacts-form'
import ContactsInfo from './contacts-info'
import ContactsBg from './contacts-bg'
// utils
import { colors } from '../../util/css-config'

const Contacts = () => {
  return (
    <SectionStyled>
      <ContainerStyled maxWidth="xl" disableGutters>
        <Grid container>
          <Grid item xs={12} md={7}>
            <ContactsInfo />
          </Grid>
          <Grid item md={5}>
            <ContactsBg />
          </Grid>
        </Grid>
        <ContactsForm />
      </ContainerStyled>
    </SectionStyled>
  )
}

const SectionStyled = styled.section`
  background: ${colors.white};

  @media (min-width: 1200px) {
    background: linear-gradient(0deg, ${colors.dark} 35%, ${colors.lightgray} 0);
    padding-bottom: 10rem;
  }
`

const ContainerStyled = styled(Container)`
  @media (min-width: 960px) {
    position: relative;
  }
`

export default Contacts
