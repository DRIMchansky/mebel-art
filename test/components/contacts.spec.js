import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { PureContactsBg as ContactsBg } from '../../src/components/contacts/contacts-bg'

describe('Contacts', () => {
  it('ContactsBg renders correctly', () => {
    const tree = renderer.create(<ContactsBg />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
