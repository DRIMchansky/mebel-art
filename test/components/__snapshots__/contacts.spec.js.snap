// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts ContactsBg renders correctly 1`] = `null`;
